WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanStatusService.kt

```diff
@@ -31,6 +31,10 @@ class ScanStatusService
     private val _scanAvailable = MutableLiveData<Boolean>()
     val scanAvailable: LiveData<Boolean> = _scanAvailable
 
+    private val _scanState = MutableLiveData<ScanStateModel>()
```

Comment:
Scan state just needs to observed from the VM. I'm using `observeForever`. I think we can use Flow   here. Lmk, I'll make the switch.

Reply:
👋 @ashiagr !

I am not sure if using `Flow` here is going to help. Since, this is a one off trigger, `scanStore.fetchScanState(...)` and `scanStore.getScanStateForSite(...)`, just like most of the api calls within our view models. It wouldn't benefit if we where to use `Flow`, would it? Am I missing something here? It is just because there is an extra layer that needs to be observed, and passed through to another observer, that you are thinking of using `Flow`, right?

What I am also seeing, is the usage of `scanStatusService.start(...)` and `scanStatusService.stop()`, which might create an intention related confusion. Without me reading through `ScanStatusService`, I though that this is indeed a `start` function that needs to be `stopped` at some point and unless it is stopped, start will continue working, producing results, but this is not the case, is it? This is actually a `trigger` function instead, and `start/stop` are being used for cleaning things up, not stopping the started service? Maybe `start/stop` should be renamed. Maybe even the `Service` naming convention is not the right one. It seems to me more like a `UseCase`. Wdyt?

Having said all of the above, I might be missing something very basic here... not sure!

Reply:
Thank you for these thoughts 🙏.

> It is just because there is an extra layer that needs to be observed, and passed through to another observer, that you are thinking of using Flow, right?

This is right, also `observeForever` is not lifecycle aware and we need to remove observer explicitly in the VM's `onCleared` but I agree to use existing conventions. 

> What I am also seeing, is the usage of scanStatusService.start(...) and scanStatusService.stop(), which might create an intention related confusion. 

I had same thoughts but I don't want to deviate too much from the `RewindStatusService` logic which is used for similar purpose and has similar `start` and `stop` methods. We might want to discuss with @jirka, @zwarm about the naming convention.

Reply:
I agree, there is a candidate for a small discussion here! 🤓

Reply:
Thank you, both! 🌟